### DISCUSSION TITLE
Customize the Rich-Editor toolbar

### DISCUSSION THREAD

**Discussion by arv-asommer** [filamentphp/filament#5711] [Help] [answered]
Hey guys,
i am facing a problem regarding the customization of the rich text editor provided by filament.
I noticed the editor being built on "trix" (https://trix-editor.org/)

I used the `resources/views/vendor` folder in order to override the rich-editor.blade.php and added my custom toolbar button.
However since the toolbar button has the attribute `[data-trix-method="setAttribute"]` I think thats what it's doing...

But I don't see any way to customize the trix textAttributes config or anything like that since trix in it of itself is not installed...

I see people online doing things like:
```js
import Trix from "trix";
 
Trix.config.textAttributes.underline = {
    style: {
        textDecoration: "underline",
    },
    inheritable: true,
    parser: (element) => {
        const style = window.getComputedStyle(element);
 
        return style.textDecoration === "underline" ||
            style.textDecorationLine === "underline";
    }
}
```

But there is no Trix object in the context of the blade file and I don't know what to do :)

I really appreciate any of your support regarding this issue!
Best regards,
Aaron S.

**Comment by sinnbeck** [comment]
Did you figure this out?

**Reply by arv-asommer** [reply, marked as answer]
Well we ended up digging deeper and implementing the tinymce editor ourself.
When then calling the `tinymce.init` function yourself you can fully customize the editor acc. to the docs (https://www.tiny.cloud/docs/tinymce/6/laravel-composer-install/)
